Framework/Algorithms/src/MergeRuns/SampleLogsBehaviour.cpp

```diff
@@ -0,0 +1,463 @@
+    Property *addeeWSProperty, const std::string name) {
+  try {
+    // If this already exists we combine the two strings.
+    auto propertyAddeeWS = addeeWS->getLog(name + LIST_SUFFIX);
+    auto propertyOutWS = outWS->mutableRun().getProperty(name + LIST_SUFFIX);
+    propertyOutWS->setValue(propertyOutWS->value() + ", " +
+                            propertyAddeeWS->value());
+  } catch (std::invalid_argument &) {
+    auto property = outWS->mutableRun().getProperty(name + LIST_SUFFIX);
+    property->setValue(property->value() + ", " + addeeWSProperty->value());
+  }
+}
+
+/**
+ * Performs the check to see if a warning should be generated because logs are
+ *different. Performs a numeric comparison if a tolerance is set and the log is
+ *a number, else performs a string comparison.
+ *
+ * @param addeeWS the workspace being merged
+ * @param addeeWSProperty the property value of the workspace being merged
+ * @param behaviour the merge behaviour struct, containing information about the
+ *merge
+ * @param addeeWSNumber a double for the addeeWS value (0 if it is not numeric)
+ * @param outWSNumber a double for the outWS value (0 if it is not numeric)
+ * @param name the name of the sample log to check
+ */
+void SampleLogsBehaviour::checkWarnProperty(const MatrixWorkspace_sptr &addeeWS,
+                                            Property *addeeWSProperty,
+                                            const SampleLogBehaviour &behaviour,
+                                            const double addeeWSNumber,
+                                            const double outWSNumber,
+                                            const std::string name) {
+  if (behaviour.isNumeric && behaviour.tolerance > 0.0) {
+    if (std::abs(addeeWSNumber - outWSNumber) > behaviour.tolerance) {
+      m_logger.warning() << generateDifferenceMessage(
+          name, addeeWS->name(), addeeWSProperty->value(),
+          behaviour.property->value());
+    }
+  } else {
+    if (behaviour.property->value().compare(addeeWSProperty->value()) != 0) {
+      m_logger.warning() << generateDifferenceMessage(
+          name, addeeWS->name(), addeeWSProperty->value(),
+          behaviour.property->value());
+    }
+  }
+}
+
+/**
+ ** Performs the check to see if an error should be generated because logs are
+ *different. Performs a numeric comparison if a tolerance is set and the log is
+ *a number, else performs a string comparison.
+ *
+ * @param addeeWS the workspace being merged
+ * @param addeeWSProperty the property value of the workspace being merged
+ * @param behaviour the merge behaviour struct, containing information about the
+ *merge
+ * @param addeeWSNumber a double for the addeeWS value (0 if it is not numeric)
+ * @param outWSNumber a double for the outWS value (0 if it is not numeric)
+ * @param name the name of the sample log to check
+ */
+void SampleLogsBehaviour::checkErrorProperty(
```

Comment:
Similar to `checkWarnProperty`, could probably be written slightly nicer. Maybe the tolerance check (which is the same in both cases) can be moved to a separate method `checkTolerance` or something like that?